altair/utils/data.py

```diff
@@ -413,7 +426,7 @@ def arrow_table_from_dfi_dataframe(dfi_df: DataFrameLike) -> pa.Table:
     # has more control over the conversion, and may have broader compatibility.
     # This is the case for Polars, which supports Date32 columns in direct conversion
     # while pyarrow does not yet support this type in from_dataframe
-    for convert_method_name in ("arrow", "to_arrow", "to_arrow_table"):
+    for convert_method_name in ("arrow", "to_arrow", "to_arrow_table", "to_pyarrow"):
```

Comment:
this change by itself isn't enough to address https://github.com/vega/altair/issues/3473 on main, because the interchange protocol is used directly - there could be an extra pyarrow conversion, but then that would worsen https://github.com/vega/altair/issues/3471